packages/modules/viewer-react/src/types.ts

```diff
@@ -134,20 +141,59 @@ export interface ViewerInitializerParams extends ViewerIModelAppOptions {
 
 export interface ConnectedViewerProps {
   iTwinId: string;
-  iModelId: string;
+  iModelId?: string;
   changeSetId?: string;
 }
-
 export interface FileViewerProps {
   /** Path to local snapshot or briefcase */
   filePath: string;
 }
 
-export interface BlankViewerProps {
-  blankConnection: BlankConnectionProps;
+export type BlankViewerProps = {
+  /** @deprecated specify location and extents instead. */
+  blankConnection?: BlankConnectionProps;
+  location?: Cartographic | EcefLocationProps;
+  extents?: Range3dProps;
+  blankConnectionViewState?: BlankConnectionViewState;
+};
+
+type BlankViewerPropsWithoutDeprecation = {
+  blankConnection?: BlankConnectionProps;
+  location?: Cartographic | EcefLocationProps;
+  extents?: Range3dProps;
   blankConnectionViewState?: BlankConnectionViewState;
+};
+
+export interface BlankConnectionInitializationProps {
+  iTwinId?: string;
+  blankConnectionProps: BlankConnectionProps;
+}
+
+export interface RequiredViewerConnectionProps {
```

Comment:
is it really _required_ if everything is optionally defined?

Reply:
Revamped.